tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/util/BCompileUtil.java

```diff
@@ -141,30 +142,49 @@ private static void runInit(BLangPackage bLangPackage, JBallerinaInMemoryClassLo
                                                                bLangPackage.packageID.name.value,
                                                                TestConstant.MODULE_INIT_CLASS_NAME);
         Class<?> initClazz = classLoader.loadClass(initClassName);
-        runOnSchedule(initClazz, bLangPackage.initFunction.name);
-        runOnSchedule(initClazz, bLangPackage.startFunction.name);
+        final Scheduler scheduler = new Scheduler(4, false);
+        runOnSchedule(initClazz, bLangPackage.initFunction.name, scheduler);
+        runOnSchedule(initClazz, bLangPackage.startFunction.name, scheduler);
+        scheduler.immortal = true;
+        new Thread(scheduler::start).start();
+
     }
 
-    private static void runOnSchedule(Class<?> initClazz, BLangIdentifier name) {
+    private static void runOnSchedule(Class<?> initClazz, BLangIdentifier name, Scheduler scheduler1) {
         String funcName = cleanupFunctionName(name);
         try {
             final Method method = initClazz.getDeclaredMethod(funcName, Strand.class);
-            Scheduler scheduler = new Scheduler(4, false);
+            Scheduler scheduler = scheduler1;
```

Comment:
redundant assignment. We can rename the `scheduler1` as `scheduler` and use instead of assigning here